kwok/cloudprovider/cloudprovider.go

```diff
@@ -77,6 +78,10 @@ func (c CloudProvider) Create(ctx context.Context, nodeClaim *v1.NodeClaim) (*v1
 			log.FromContext(ctx).Error(err, "failed creating node from nodeclaim")
 		}
 	}()
+	nodeClassReady := nodeClass.StatusConditions().Get(status.ConditionReady)
```

Comment:
Adding support for Kwok nodeclass that are not ready: https://github.com/kubernetes-sigs/karpenter/pull/2193/files#diff-ec78a380674cc833d5255f7b850e5ae25337cd6099fed376cc7e177e449b19c8R260